pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -383,11 +347,9 @@ public CompletableFuture<Void> abortTxn(TxnID txnID, long lowWaterMark) {
                     @Override
                     public void addComplete(Position position, ByteBuf entryData, Object ctx) {
                         synchronized (TopicTransactionBuffer.this) {
-                            aborts.put(txnID, (PositionImpl) position);
-                            updateMaxReadPosition(txnID);
-                            changeMaxReadPositionAndAddAbortTimes.getAndIncrement();
-                            clearAbortedTransactions();
-                            takeSnapshotByChangeTimes();
+                            PositionImpl maxReadPosition = updateMaxReadPosition(txnID);
+                            snapshotAbortedTxnProcessor.appendAbortedTxn(txnID, maxReadPosition);
```

Comment:
These two ops must guarantee atomicity with take snapshot, otherwise, transient may be wrong

Reply:
I think we can make `updateMaxReadPosition` after `appendAbortedTxn`.